paddle/fluid/operators/fused_elemwise_activation_op.h

```diff
@@ -385,40 +277,79 @@ class FusedElemwiseActivationKernel : public framework::OpKernel<T> {
+    auto outputs = ctx.MultiOutput<framework::Tensor>("Out");
 
-    RunFunctors<DeviceContext, T>(ctx, &in_x, &in_y, &output);
+    if (!ctx.Attr<bool>("keep_intermediate_value")) {
+      outputs.push_back(nullptr);
+    }
+
+    RunFunctors<DeviceContext, T>(ctx, in_x, in_y, &outputs);
   }
 };
 
 template <typename DeviceContext, typename T>
 class FusedElemwiseActivationGradKernel : public framework::OpKernel<T> {
  public:
   void Compute(const framework::ExecutionContext &ctx) const override {
-    auto &in_x = detail::Ref(ctx.Input<framework::Tensor>("X"),
-                             "Cannot get input tensor %s, variable name = %s",
-                             "X", ctx.op().Input("X"));
-    auto &in_y = detail::Ref(ctx.Input<framework::Tensor>("Y"),
-                             "Cannot get input tensor %s, variable name = %s",
-                             "Y", ctx.op().Input("Y"));
-    auto &in_out = detail::Ref(ctx.Input<framework::Tensor>("Out"),
-                               "Cannot get input tensor %s, variable name = %s",
-                               "Out", ctx.op().Input("Out"));
-    auto &in_out_grad =
-        detail::Ref(ctx.Input<framework::Tensor>(framework::GradVarName("Out")),
-                    "Cannot get input tensor %s, variable name = %s",
-                    framework::GradVarName("Out"),
-                    ctx.op().Input(framework::GradVarName("Out")));
+    auto x = ctx.Input<framework::Tensor>("X");
+    auto y = ctx.Input<framework::Tensor>("Y");
+
+    auto &in_outs = ctx.MultiInput<framework::Tensor>("Out");
+    auto &in_outs_grad =
+        ctx.MultiInput<framework::Tensor>(framework::GradVarName("Out"));
 
     framework::Tensor *x_grad =
         ctx.Output<framework::Tensor>(framework::GradVarName("X"));
     framework::Tensor *y_grad =
         ctx.Output<framework::Tensor>(framework::GradVarName("Y"));
 
-    RunGradFunctors<DeviceContext, T>(ctx, &in_x, &in_y, &in_out, &in_out_grad,
-                                      x_grad, y_grad);
+    if (ctx.Attr<bool>("recomputation")) {
+      PADDLE_ENFORCE(
+          x != nullptr,
+          "The recomputation is opened, so Input(X) should not be absent.");
+      PADDLE_ENFORCE(
+          y != nullptr,
+          "The recomputation is opened, so Input(Y) should not be absent.");
+    } else {
+      PADDLE_ENFORCE_GE(in_outs.size(), 1,
+                        "The recomputation is disabled,"
+                        "so the Input('Out') should not be empty.");
+    }
+
+    framework::Tensor *in_x, *in_y;
+    framework::Tensor *in_out, *in_intermediate_out;
+    // If functor_list contains elementwise_add, the backward doesn't use
+    // in_x,in_y and in_outs.
+    auto functor_list = ctx.Attr<std::vector<std::string>>("functor_list");
+    if (functor_list[0] == "elementwise_add" ||
+        functor_list[1] == "elementwise_add") {
+      in_x = const_cast<framework::Tensor *>(in_outs_grad[0]);
```

Comment:
should you use Output() above to avoid `const_cast` here and below?

Reply:
In some situation, we only want to get `dx` or `dy` but not both them, so we use Output() here, we should check whether `dx` or `dy` is nullptr, beside the number of elements of dx and dy is different.